jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/JaxRsRequest.java

```diff
@@ -108,8 +108,13 @@ public String getQueryParam(String key) {
     public String getQueryParam(String key, String defaultValue) {
         if (queryParameters == null) {
             QueryStringDecoder queryStringDecoder = new QueryStringDecoder(req.uri());
-            queryParameters = queryStringDecoder.parameters();
+            try {
+                queryParameters = queryStringDecoder.parameters();
+            } catch (IllegalArgumentException e) {
+                throw new WebException(HttpResponseStatus.BAD_REQUEST, "invalidhexbyte", "Failed to decode query params due to invalid two digit hexadecimal");
```

Comment:
How do we know that the IllegalArgumentException is caused by what we state and assume here?

Reply:
I made the assumption that IllegalArgumentException is thrown if and only if the hexbyte decoding fails. I could not find other places where an IllegalArgumentException would be thrown for some query param values other than the hexbyte decoding part of the code. That said, I do not think we should pass the exception message to WebException but we can change the message to something more general, like "Failed to decode query parameters" and the "invalidhexbyte" can be changed to "querydecodefailure".

Reply:
Will any of that (except the status code) be exposed in the http response? If so, I don't think we should include it in the WebException, but log the IllegalArgumentException instead.